pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -175,51 +182,66 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 			r.status.OnCRNotFound()
 			return reconcile.Result{}, nil
 		}
-		r.status.SetDegraded(msg, err.Error())
-		reqLogger.Error(err, fmt.Sprintf("An error occurred when querying the APIServer resource: %s", msg))
+		r.SetDegraded(operatorv1.ResourceReadError, fmt.Sprintf("An error occurred when querying the APIServer resource: %s", msg), err, reqLogger)
 		return reconcile.Result{}, err
 	}
 	r.status.OnCRFound()
 	reqLogger.V(2).Info("Loaded config", "config", instance)
 
+	//Set the meta info in the tigerastatus like observedGenerations
+	if instance != nil {
+		defer r.status.SetMetaData(&instance.ObjectMeta)
+	}
+
+	// Changes for updating apiserver status conditions
+	if request.Name == ResourceName && request.Namespace == "" {
```

Comment:
I believe we no longer end the reconcile methods pre-emptively.

Reply:
Yes.. we are not ending the reconcile pre-emptively..  I have checked in all the place again.. Let me know if i have missed it in any place..